In [1]:
import numpy as np
import my_functions as f
from scipy.stats import norm
from scipy.stats import chi2

In [2]:
normal_distribution_file = open("N.txt", "r")

data = np.array(normal_distribution_file.read().replace('\n', ' ').split(' ')).astype("float64")


In [3]:
X_mean = np.mean(data)

In [4]:
X_mean

1.02726

$\overline{X}$ = 1.0276

In [5]:
D = np.mean(data * data) - X_mean * X_mean

In [6]:
D

0.8876415124000001

$\frac{1}{50} * \sum_1^{50} (X_i - \overline{X})^2$ = 0.8876415124000001

In [7]:
S = data.size * D / (data.size - 1)

In [8]:
S

0.9057566453061227

$S^2_0$= 0.9057566453061227

1
a)
$$P(\sqrt{n} \frac{\overline{X} - a}{\sigma} < A) \in N_{0, 1}$$
$$P(-A \leqslant	\sqrt{n} \frac{\overline{X} - a}{\sigma} < A) = \Phi_{0, 1}(A) - \Phi_{0, 1}(-A) = 2*\Phi_{0, 1}(A) - 1 = 1 - \epsilon
$$
Найдем A:

In [9]:
epsilon = 0.16
a = 1
sigma2 = 1.1
A = f.find_quantile((2 - epsilon) / 2, norm.cdf)

In [10]:
A

1.40380859375

A = 1.40380859375
$$ -A <	\sqrt{n} \frac{\overline{X} - a}{\sigma} < A $$
$$ \overline{X} - \frac{A*\sigma}{\sqrt{n}} < a  < \overline{X} + \frac{A*\sigma}{\sqrt{n}}$$

In [11]:
X_mean - A * np.sqrt(sigma2)/np.sqrt(data.size)

0.8190415366171873

In [12]:
X_mean + A * np.sqrt(sigma2)/np.sqrt(data.size)

1.2354784633828129

    Ответ: $\overline{X}$ = 1.02726,
 A = 1.40380859375,
 P(0.8190415 < a < 1.2354784) = 1 - $\epsilon$

b)
$\sqrt{n} \frac{\overline{X} - a}{\sigma}$  сходится к $N_{0,1}$
$\frac{\sigma}{S_0}$ сходится к 1

тогда $\sqrt{n} \frac{\overline{X} - a}{\sigma} * \frac{\sigma}{S_0}$  также  сходится к $N_{0,1}$

$$P(-A \leqslant	\sqrt{n} \frac{\overline{X} - a}{S_0} < A) = \Phi_{0, 1}(A) - \Phi_{0, 1}(-A) = 2*\Phi_{0, 1}(A) - 1 = 1 - \epsilon
$$
A такой же, как и в прошлой задаче, найдем а

A = 1.40380859375
$$ -A <	\sqrt{n} \frac{\overline{X} - a}{S} < A $$
$$ \overline{X} - \frac{A*S_0}{\sqrt{n}} < a  < \overline{X} + \frac{A*S_0}{\sqrt{n}}$$

In [13]:
X_mean + A * np.sqrt(S)/np.sqrt(data.size)

1.2162020656511099

In [14]:
X_mean - A * np.sqrt(S)/np.sqrt(data.size)


0.8383179343488901

Ответ: $S^2_0$ = 0.9057566453061227,
 A = 1.40380859375,
 P(0.838317< a < 1.216202) = 1 - $\epsilon$

c)
$\sum_{i = 1}^n (\frac{X_i - a}{\sigma})^2 \sim \chi^2_n$
$P(h_{\frac{\epsilon}{2}} < \sum_{i = 1}^n (\frac{X_i - a}{\sigma})^2 < h_{1- \frac{\epsilon}{2}}) = 1 - \epsilon $

In [15]:
he2 = f.find_quantile(epsilon/2, lambda x: chi2.cdf(x, data.size))
h1_e2 = f.find_quantile(1 - epsilon/2, lambda x: chi2.cdf(x, data.size))

In [16]:
(he2, h1_e2)

(36.68212890625, 64.697265625)

$$h_{\frac{\epsilon}{2}} < \sum_{i = 1}^n (\frac{X_i - a}{\sigma})^2 < h_{1- \frac{\epsilon}{2}}$$
$$\frac{\sum_{i = 1}^n (X_i - a)^2}{h_{1- \frac{\epsilon}{2}}} < \sigma^2 < \frac{\sum_{i = 1}^n (X_i - a)^2}{h_{\frac{\epsilon}{2}}} $$
Найдем границы для $\sigma^2$

In [17]:
(np.sum(np.power(data - a, 2))/h1_e2, np.sum(np.power(data - a, 2))/he2)

(0.6865704534943395, 1.2109229296239599)

Ответ:
$h_{\frac{\epsilon}{2}}$ = 36.716461181640625,
$h_{1 - \frac{\epsilon}{2}}$ = 64.61143493652344
P(0.6874825028052546 < $\sigma^2$ < 1.2097906380534025) = 1 - $\epsilon$

d)
$\frac{(n-1)*S^2_0}{\sigma^2} \sim \chi^2_{n-1}$
$P(h_{\frac{\epsilon}{2}} < \frac{(n-1)*S^2_0}{\sigma^2} < h_{1- \frac{\epsilon}{2}}) = 1 - \epsilon $

$$\frac{(n-1)*S^2_0}{h_{1- \frac{\epsilon}{2}}} < \sigma^2 < \frac{(n-1)*S^2_0}{h_{\frac{\epsilon}{2}}} $$

In [18]:
((data.size - 1) * S / h1_e2, (data.size - 1) * S / he2)

(0.6859961575076228, 1.2099100282164394)

Ответ:
$h_{\frac{\epsilon}{2}}$ = 36.716461181640625,
$h_{1 - \frac{\epsilon}{2}}$ = 64.61143493652344
P(0.68690744391612 < $\sigma^2$ <  1.208778683774471) = 1 - $\epsilon$